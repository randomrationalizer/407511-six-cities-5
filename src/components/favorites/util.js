export const getCitiesFromOffers = (offers) => {
  const cities = offers.slice().map((offer) => offer.city.name).sort();
  return [...new Set(cities)];
};
