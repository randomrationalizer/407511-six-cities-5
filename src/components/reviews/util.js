const REVIEW_MIN_LENGTH = 50;
const REVIEW_MAX_LENGTH = 300;
const RATING_MIN = 1;
const RATING_MAX = 5;

export const checkCommentValidity = (comment) => {
  return comment.length >= REVIEW_MIN_LENGTH && comment.length <= REVIEW_MAX_LENGTH ? true : false;
};

export const checkRatingValidity = (rating) => {
  return rating >= RATING_MIN && rating <= RATING_MAX ? true : false;
};
