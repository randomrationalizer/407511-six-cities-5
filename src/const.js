export const cities = [
  {
    name: `Paris`,
    coords: {
      latitude: 48.864716,
      longitude: 2.349014
    }
  },
  {
    name: `Cologne`,
    coords: {
      latitude: 50.935173,
      longitude: 6.953101
    }
  },
  {
    name: `Brussels`,
    coords: {
      latitude: 50.8505,
      longitude: 4.3488
    }
  },
  {
    name: `Amsterdam`,
    coords: {
      latitude: 52.38333,
      longitude: 4.9
    }
  },
  {
    name: `Hamburg`,
    coords: {
      latitude: 53.551086,
      longitude: 9.993682
    }
  },
  {
    name: `Dusseldorf`,
    coords: {
      latitude: 51.233334,
      longitude: 6.783333
    }
  }
];

export const OfferType = {
  MAIN: `cities`,
  NEARBY: `near-places`,
  FAVORITES: `favorites`
};

export const MapType = {
  MAIN: `cities`,
  PROPERTY: `property`
};

export const SortType = {
  DEFAULT: `DEFAULT`,
  PRICE_UP: `PRICE_UP`,
  PRICE_DOWN: `PRICE_DOWN`,
  RATING_DOWN: `RATING_DOWN`
};
