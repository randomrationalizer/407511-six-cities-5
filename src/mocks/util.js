import configureStore from "redux-mock-store";

export const noop = () => {};

export const getMockStore = (store) => {
  const storeConfig = configureStore();
  return storeConfig(store);
};

export const nope = () => {
  return Promise.resolve({});
};
