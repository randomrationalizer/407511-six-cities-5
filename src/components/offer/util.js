import dayjs from "dayjs";
import {OfferType} from "../../const";


export const capitalize = (str) => {
  return str.charAt(0).toUpperCase() + str.slice(1);
};

// Возвращает дату в виде строки в формате: 2019-04-24
export const formatDate = (date) => {
  return dayjs(date).format(`YYYY-MM-DD`);
};

// Возвращает дату в виде строки в формате: January 2020
export const humanizeDate = (date) => {
  return dayjs(date).format(`MMMM YYYY`);
};

// Сравнение для сортировки отзывов в хронологическом порядке
export const sortByDate = (a, b) => {
  return dayjs(a.date) - dayjs(b.date);
};

export const getDescriptionSentences = (text) => {
  let description = text.split(`. `);
  description[description.length - 1] = description[description.length - 1].replace(`.`, ``);

  return description;
};

export const isFavoritesCard = (type) => {
  return type === OfferType.FAVORITES;
};

export const isMainPageCard = (type) => {
  return type === OfferType.MAIN;
};
