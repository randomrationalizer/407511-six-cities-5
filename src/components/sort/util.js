import {SortType} from "../../const";

export const sortTypeToTitle = {
  [SortType.DEFAULT]: `Popular`,
  [SortType.PRICE_UP]: `Price: low to high`,
  [SortType.PRICE_DOWN]: `Price: high to low`,
  [SortType.RATING_DOWN]: `Top rated first`
};
