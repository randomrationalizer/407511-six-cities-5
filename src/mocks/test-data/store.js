import configureStore from "redux-mock-store";
import {NameSpace} from "../../store/root-reducer";
import mockOffers from "./offers";
import mockReviews from "./reviews";
import mockNearbyOffers from "./nearby-offers";
import mockCurrentOffer from "./current-offer";
import mockUserInfo from "./user-info";
import mockFavorites from "./favorites";
import mockCities from "./cities";
import {SortType, AuthorizationStatus, DEFAULT_CITY} from "../../const";


export default {
  [NameSpace.APP_DATA]: {
    currentCity: DEFAULT_CITY,
    allOffers: mockOffers,
    cities: mockCities,
    currentSort: SortType.DEFAULT
  },
  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.AUTH,
    isAuthRequestComplete: true,
    userInfo: mockUserInfo,
    favorites: mockFavorites
  },
  [NameSpace.CURRENT_OFFER]: {
    offer: mockCurrentOffer,
    reviews: mockReviews,
    nearbyOffers: mockNearbyOffers
  },
  [NameSpace.LOAD_STATUS]: {
    isOffersLoaded: true,
    isFavoritesLoaded: true,
    isCurrentOfferLoaded: true,
    errorMessage: null,
    isLoadFinished: true
  }
};
