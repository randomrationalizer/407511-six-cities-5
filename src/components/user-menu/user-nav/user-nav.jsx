import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import ProfileLink from "../profile-link/profile-link";
import ProfileLinkUnauthorized from "../profile-link-unauthorized/profile-link-unauthorized";
import {AuthorizationStatus} from "../../../const";
import {getAuthorizationStatus} from "../../../store/selectors";


const UserNav = (props) => {
  const {authorizationStatus} = props;
  const isAuthorized = authorizationStatus === AuthorizationStatus.AUTH;

  return (
    <nav className="header__nav">
      <ul className="header__nav-list">
        <li className="header__nav-item user">
          {isAuthorized ?
            <ProfileLink />
            :
            <ProfileLinkUnauthorized />
          }
        </li>
      </ul>
    </nav>
  );
};

UserNav.propTypes = {
  authorizationStatus: PropTypes.string.isRequired
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state)
});

export {UserNav};
export default connect(mapStateToProps, null)(UserNav);

