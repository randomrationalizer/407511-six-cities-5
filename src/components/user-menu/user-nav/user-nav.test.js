import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import {UserNav} from "./user-nav";
import store from "../../../mocks/test-data/store";
import {getMockStore} from "../../../mocks/util";
import {AuthorizationStatus} from "../../../const";


const mockStore = getMockStore(store);


describe(`Should UserNav component renders correctly`, () => {
  it(`if user has been authenticate`, () => {
    const tree = renderer
      .create(
          <Provider store={mockStore}>
            <BrowserRouter>
              <UserNav
                authorizationStatus={AuthorizationStatus.AUTH}
              />
            </BrowserRouter>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`if user is not authenticated`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <UserNav
              authorizationStatus={AuthorizationStatus.NO_AUTH}
            />
          </BrowserRouter>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
