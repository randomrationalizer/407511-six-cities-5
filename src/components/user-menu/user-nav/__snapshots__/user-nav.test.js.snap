// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should UserNav component renders correctly if user has been authenticate 1`] = `
<nav
  className="header__nav"
>
  <ul
    className="header__nav-list"
  >
    <li
      className="header__nav-item user"
    >
      <a
        className="header__nav-link header__nav-link--profile"
        href="/favorite"
        onClick={[Function]}
      >
        <div
          className="header__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="User avatar"
            className="user__avatar"
            height={54}
            src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg"
            width={54}
          />
        </div>
        <span
          className="header__user-name user__name"
        >
          Oliver.conner@gmail.com
        </span>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Should UserNav component renders correctly if user is not authenticated 1`] = `
<nav
  className="header__nav"
>
  <ul
    className="header__nav-list"
  >
    <li
      className="header__nav-item user"
    >
      <a
        className="header__nav-link header__nav-link--profile"
        href="/login"
        onClick={[Function]}
      >
        <div
          className="header__avatar-wrapper user__avatar-wrapper"
        />
        <span
          className="header__login"
        >
          Sign in
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
